Userland/Libraries/LibWeb/HTML/WebViewHints.cpp

```diff
@@ -6,7 +6,122 @@
 
 #include <LibIPC/Decoder.h>
 #include <LibIPC/Encoder.h>
+#include <LibWeb/HTML/Numbers.h>
 #include <LibWeb/HTML/WebViewHints.h>
+#include <LibWeb/Page/Page.h>
+#include <LibWeb/PixelUnits.h>
+
+namespace Web::HTML {
+
+static void set_up_browsing_context_features(WebViewHints& target, TokenizedFeature::Map const& tokenized_features, Page const& page);
+
+WebViewHints WebViewHints::from_tokenised_features(TokenizedFeature::Map const& tokenized_features, Page const& page)
+{
+    WebViewHints hints;
+    hints.popup = check_if_a_popup_window_is_requested(tokenized_features) == TokenizedFeature::Popup::Yes;
+    set_up_browsing_context_features(hints, tokenized_features, page);
+    return hints;
+}
+
+// https://drafts.csswg.org/cssom-view/#set-up-browsing-context-features
+static void set_up_browsing_context_features(WebViewHints& target, TokenizedFeature::Map const& tokenized_features, Page const& page)
+{
+    // 1. Let x be null.
+    Optional<CSSPixels> x;
+
+    // 2. Let y be null.
+    Optional<CSSPixels> y;
+
+    // 3. Let width be null.
+    Optional<CSSPixels> width;
+
+    // 4. Let height be null.
+    Optional<CSSPixels> height;
+
+    auto screen_rect = page.web_exposed_screen_area();
+
+    // 5. If tokenizedFeatures["left"] exists:
+    if (auto left = tokenized_features.get("left"sv); left.has_value()) {
+        // 1. Set x to the result of invoking the rules for parsing integers on tokenizedFeatures["left"].
+        // 2. If x is an error, set x to 0.
+        x = parse_integer(*left).value_or(0);
+
+        // 3. Optionally, clamp x in a user-agent-defined manner so that the window does not move outside the Web-exposed available screen area.
+        x = min(*x, screen_rect.x());
```

Comment:
:thinking: wonder if we should also make sure it doesn't go off the right side of the screen, i.e. limit to `screen_width - screen_x`. Similar with `y` below.

Reply:
I do wonder, how much of this is going to be handled by the GUI framework on the UI process side in the end? but I can try to hit detect the sides of the box in the size cases and adjust x/y.

Reply:
Yeah good point, I dunno - my main concern was a website throwing the window wayyy off the right of the screen where users can't get it, so I think we're good as long as we prevent that, then let the UI kits handle whatever else they want